### PR TITLE
editoast: refactor main.rs db_pool

### LINKED ISSUE #5703: Refacto main.rs db_pool

### PR DESCRIPTION
**Pull request by anisometropie**
closes #5703

### CODE REVIEW

File: editoast/src/main.rs [hunk 1]
```diff
@@ -81,6 +81,7 @@ async fn main() {
 async fn run() -> Result<(), Box<dyn Error + Send + Sync>> {
     let client = Client::parse();
     let pg_config = client.postgres_config;
+    let db_pool = Data::new(get_pool(pg_config.url()?, pg_config.pool_size));
```
Comment (leovalais) [inline editoast/src/main.rs]: Some commands do not require a db_pool (osm, openapi, some search commands, maybe more). Match against these variants before creating the connection pool.

**Review by leovalais — CHANGES_REQUESTED** [review]
Thanks!

File: editoast/src/main.rs [hunk 2]
```diff
@@ -81,6 +80,10 @@ async fn main() {
 async fn run() -> Result<(), Box<dyn Error + Send + Sync>> {
     let client = Client::parse();
     let pg_config = client.postgres_config;
+    let create_db_pool = || match pg_config.url() {
+        Ok(url) => Ok(Data::new(get_pool(url, pg_config.pool_size))),
+        Err(e) => Err(e),
+    };
```
Comment (leovalais) [inline editoast/src/main.rs]:
```suggestion
    let create_db_pool = || Ok::<_, Box<dyn Error + Send + Sync>>(Data::new(get_pool(pg_config.url()?, pg_config.pool_size)));
```

**Review by leovalais — CHANGES_REQUESTED** [review]
Almost good!

File: editoast/src/main.rs (hunk 2 above)
Comment (anisometropie) [reply editoast/src/main.rs]: done. 

the linter wants to have the {} for the closure though

```
let create_db_pool = || {
        Ok::<_, Box<dyn Error + Send + Sync>>(Data::new(get_pool(
            pg_config.url()?,
            pg_config.pool_size,
        )))
};
```

**Review by leovalais — APPROVED** [review]

File: editoast/src/main.rs [hunk 3]
```diff
@@ -266,7 +271,7 @@ async fn runserver(
             .wrap(Logger::new(actix_logger_format).log_target("actix_logger"))
             .app_data(json_cfg.clone())
             .app_data(payload_config.clone())
-            .app_data(Data::new(pool.clone()))
+            .app_data(Data::new(db_pool.clone()))
```
Comment (leovalais) [inline editoast/src/main.rs]:
```suggestion
            .app_data(db_pool)
```

**Pull request merged**